scripts/Diffraction/isis_powder/routines/common.py

```diff
@@ -207,6 +207,9 @@ def get_first_run_number(run_number_string):
     """
     run_numbers = generate_run_numbers(run_number_string=run_number_string)
     if isinstance(run_numbers, list):
+        if not run_numbers:
+            raise RuntimeError("Attempted to load empty set of workspaces. Please input at least one valid run number")
```

Comment:
`generate_run_numbers` looks like always returns a list, so the `isinstance` call is redundant. So is the inner check. perhaps just `if len(run_numbers) == 0:`

Reply:
the inner check isn't redundant - you can fire it off by passing `""` as your run number. good spot on `isinstance` though